packages/react/src/List/OrderedList.tsx

```diff
@@ -0,0 +1,19 @@
+import React, {PropsWithChildren} from 'react'
+import clsx from 'clsx'
+import type {BaseProps} from '../component-helpers'
+import {ListContext} from './listContext'
+import {ListItem} from './ListItem'
+
+import styles from './List.module.css'
+
+export type OrderedListProps = PropsWithChildren<BaseProps<HTMLOListElement>>
+
+function Root({children, ...props}: OrderedListProps) {
+  return (
+    <ol className={clsx(styles.List, props.className)} {...props}>
+      <ListContext.Provider value={{variant: ''}}>{children}</ListContext.Provider>
```

Comment:
Looking at this fresh, I wouldn't know that an empty string means it's for ordered lists. There should be type safety around this. Can we extend the switch case to mention numeric and handle it in a truthy way, rather than falsey via empty string.

Reply:
I think this might be a preference thing as the user does not have access to this in any way. It accurately displays that no variant is supplied which refers the the context of leading visuals. I think that empty string is more accurate to its functionality than passing numeric would be.

Reply:
> I think this might be a preference thing as the user does not have access to this in any way. 

Yep, feedback is coming from maintainability perspective, not user.

> I think that empty string is more accurate to its functionality than passing numeric would be

In future we may need to extend support to definition lists, or other type of lists. The switch works well for that scenario, but there's an assumption that `'' === 'numeric`, which isn't immediately clear for anyone that has to implement this in future. I would also assume some type safety here, given we have it for everything else.

Reply:
I think this is another reason if we want to improve user experience and developer experience these should in fact be one component rather than 2. This would simplify use and would mitigate these issues that come down to preference and maintainability.